src/transformers/pipelines/image_segmentation.py

```diff
@@ -0,0 +1,154 @@
+                )
+        elif isinstance(image, Image.Image):
+            pass
+        else:
+            raise ValueError(
+                "Incorrect format used for image. Should be a URL linking to an image, a local path, or a PIL image."
+            )
+        image = image.convert("RGB")
+        return image
+
+    def _sanitize_parameters(self, **kwargs):
+        postprocess_kwargs = {}
+        if "threshold" in kwargs:
+            postprocess_kwargs["threshold"] = kwargs["threshold"]
+        if "subtask" in kwargs:
+            postprocess_kwargs["subtask"] = kwargs["subtask"]
+        return {}, {}, postprocess_kwargs
+
+    def __call__(self, *args, **kwargs) -> Union[Predictions, List[Prediction]]:
+        """
+        Perform segmentation (detect maskss & classes) in the image(s) passed as inputs.
+
+        Args:
+            images (:obj:`str`, :obj:`List[str]`, :obj:`PIL.Image` or :obj:`List[PIL.Image]`):
+                The pipeline handles three types of images:
+
+                - A string containing an HTTP(S) link pointing to an image
+                - A string containing a local path to an image
+                - An image loaded in PIL directly
+
+                The pipeline accepts either a single image or a batch of images. Images in a batch must all be in the
+                same format: all as HTTP(S) links, all as local paths, or all as PIL images.
+            threshold (:obj:`float`, `optional`, defaults to 0.9):
+                The probability necessary to make a prediction.
+            subtask (:obj:`str`, `optional`, defaults to None):
+                Type of image segmentation: a value from set (semantic, panoptic, part, instance, part-aware)
+
+        Return:
+            A dictionary or a list of dictionaries containing the result. If the input is a single image, will return a
+            dictionary, if the input is a list of several images, will return a list of dictionaries corresponding to
+            each image.
+
+            The dictionaries contain the following keys:
+
+            - **png_string** (:obj:`str`) -- base64 string of a PNG image that contain masks information. Per-pixel
+              segment ids are stored as the PNG image. Each segment is assigned a unique id. Unlabeled pixels (void)
+              are assigned a value of 0. Note that when you load the PNG as an RGB image, you will need to compute the
+              ids via ids=R+G*256+B*256^2 (inspired by https://cocodataset.org/#format-data).
+            - **segments_info** (:obj:`List[dict]`) -- list that contains labels for segments.
+
+            **segments_info** contains dictionaries with the following keys:
+
+            - **id** (:obj:`int`) -- The id attributed for segment.
+            - **label** (:obj:`str`) -- The class label identified by the model.
+            - **score** (:obj:`float`) -- The score attributed by the model for that label.
```

Comment:
Am I correct to assume that you are proposing the output to be: 
```py
List(Dict(
    mask: str, // base64 str
    label: float,
    score: number,
    id: int
))
```
?

Reply:
Exactly ! I might be missing some subtleties but it should work.

(For `part-aware` for instance you need some kind of tree structure, but I think `parent?: int` could work in that case where `parent` will be the index of the parent mask).

Reply:
If I'm not mistaken (cc: @NielsRogge), the current DetrForSegmentation implements `panoptic` segmentation, not `part-aware panoptic`. Therefore, should I just put -1 the parent indeces for DetrForSegmentation atm?
```py
List(Dict(
    mask: str, // base64 str
    label: float,
    score: int,
    parent: int // -1 for all for DetrForSegmentation
))
# removing `id` field from dict since it is no longer needed
```
And in the future, we might have segmentation architectures that can do `part-aware panoptic`, in which case `parent` fields will come in handy. WDYT @Narsil

Reply:
Yes, let's not add what's not necessary, we know how to add that's enough (a commentary maybe for future implementors but that's the extent of where I would go.

Reply:
pushed the suggested changes:
1. Change the output shape to the one discussed above
2. Remove subtask argument, thereby making tests simpler